tests/queries/tests.py

```diff
@@ -1314,7 +1314,7 @@ def test_ticket7107(self):
     def test_ticket8683(self):
         # An error should be raised when QuerySet.datetimes() is passed the
         # wrong type of field.
-        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateTimeField."):
+        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateField or DateTimeField."):
```

Comment:
If I understand correctly, this test shows that `queryset.datetimes()` now accepts a `DateField`.

We should check why I introduced this new method (in 1.6 if memory serves). Generally speaking the idea was that `dates()` should only accept dates and that `datetimes()` should only accept datetimes. I'm not sure why I couldn't do it just with the `dates()` API and that's what we need to check to determine the correct course of action.